### ISSUE TITLE
Parser meta lookups don't work for INSERT .. SELECT .. RETURNING

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13503]
Given this schema:

```sql
create table t (i int, j int);
```

The parse meta lookups work for this `INSERT .. VALUES .. RETURNING`:

```
> insert into t (i, j) values (1, 2) returning t.i;
insert into t (i, j) values (1, 2) returning t.i;

> insert into t (i, j) values (1, 2) returning t.x;
Unknown field identifier: [1:46] insert into t (i, j) values (1, 2) returning [*]t.x;
```

But not for `INSERT .. SELECT .. RETURNING`:

```
> insert into t (i, j) select 1, 2 returning t.i;
Unknown field identifier: [1:44] insert into t (i, j) select 1, 2 returning [*]t.i;
```

**Comment by lukaseder** [comment]
The problem is related to the fix of #10954

**Comment by lukaseder** [comment]
OK, no, the problem is this:

```sql
create table U (COL1 int, COL2 varchar(10));
create table O (X int, Y varchar(10));
insert into U (COL1) select COL1 FROM O
```

The query shouldn't work, because even if in the outer scope, `COL1` is defined, it must not be defined in the `SELECT` scope. So, the parser just ends the `INSERT` scope, starting a new one. When in fact, it should restore the `INSERT` scope again after the `SELECT` statement.

**Comment by lukaseder** [comment]
Works correctly now:

```
> insert into t (i, j) select 1, 2 returning t.i;
insert into t (i, j) select 1, 2 returning t.i;

> insert into t (i, j) select 1, 2 returning t.y;
Unknown field identifier: [1:44] insert into t (i, j) select 1, 2 returning [*]t.y;
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13504), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13505)
